examples/draw_M_z_from_hmf.ipynb

```diff
@@ -0,0 +1,863 @@
+    "logm = 15.\n",
+    "res_ccl = t08_vec_ccl(logm,z_arr)\n",
+    "res_nc = t08_vec_nc(logm,z_arr)\n",
+    "plt.plot(z_arr, np.abs(res_ccl/res_nc -1)*100);\n",
+    "plt.xlabel(\"Redshift\", size=14);\n",
+    "plt.ylabel(\"Relative difference [%]\", size=14);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "beed29aa-f030-4e4f-8b0e-71e6aa36b6b7",
+   "metadata": {},
+   "source": [
+    "### Acceptance-rejection method to sample (M,z) from the mass function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "39352c6b-4a61-40c9-92ce-436f16fc8c6f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def bivariate_draw(pdf, N=1000,logm_min=14.,logm_max=15., zmin=0.01, zmax=1, Ngrid_m=30, Ngrid_z=30):  \n",
+    "    \"\"\"  \n",
+    "    Uses the rejection method for generating random numbers derived from an arbitrary   \n",
+    "    probability distribution. \n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    pdf : func\n",
+    "      2d distribution function to sample\n",
+    "    N : int\n",
+    "      number of points to generate  \n",
+    "    log_min,logm_max : float \n",
+    "      log10 mass range  \n",
+    "    zmin,zmax : float \n",
+    "      redshift range  \n",
+    "\n",
+    "    Returns:   \n",
+    "    --------\n",
+    "    ran_logm : list\n",
+    "        accepted logm values  \n",
+    "    ran_z : list\n",
+    "        accepted redshift values\n",
+    "    acceptance : float\n",
+    "        acceptance ratio of the method\n",
+    "    \"\"\"  \n",
+    "\n",
+    "    # maximum value of the pdf over the mass and redshift space.\n",
+    "    # the pdf is not monotonous in mass and redshift, so we need\n",
+    "    # to find the maximum numerically. Here we scan the space\n",
+    "    # with a regular grid and use the maximum value.\n",
+    "    # Accuracy of the results depends on Ngrid_m and Ngrid_z \n",
+    "    # This should probably be improved\n",
+    "    \n",
+    "    logM_arr = np.linspace(logm_min, logm_max, Ngrid_m)  \n",
+    "    z_arr = np.logspace(np.log10(zmin), np.log10(zmax), Ngrid_z)  \n",
+    "    p = []\n",
+    "    for logM in logM_arr:\n",
+    "        for z in z_arr:\n",
+    "            p.append(pdf(logM, z))  \n",
+    "    pmax = np.max(p)\n",
+    "    pmin = np.min(p)\n",
+    "    # Counters  \n",
+    "    naccept = 0  \n",
+    "    ntrial = 0  \n",
+    "\n",
+    "    # Keeps drawing until N points are accepted  \n",
+    "    ran_logm = [] # output list of random numbers  \n",
+    "    ran_z = [] # output list of random numbers  \n",
+    "    while naccept < N:  \n",
+    "        # draw (logm,z) from uniform distribution\n",
+    "        # draw p from uniform distribution\n",
+    "        logm = np.random.uniform(logm_min,logm_max) # x'  \n",
+    "        z = np.random.uniform(zmin,zmax) # x'  \n",
+    "        p = np.random.uniform(pmin,pmax) # y'  \n",
```

Comment:
I changed the pdf to start from pmin instead of 0

Reply:
I reverted to 0 as the sanity check was not passing when using pmin. I remember having to convince myself that this was what needed to be used, but cannot think of the reason now. Definitely needs to go back to it, but the behaviour starting from 0 is the right one.

Reply:
If you sort a random number starting from `pmin` points around `P(m, z)=pmin` will never be selected. You want them selected `pmin/pmax` of the time.